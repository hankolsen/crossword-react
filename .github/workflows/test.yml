name: melodikryzzet-tests

on: 
  push:
    branches:
      - 'master'
  pull_request:


jobs:
  test:
    name: 'Node ${{ matrix.node }}'
    runs-on: ubuntu-latest
    strategy:
      matrix:
        node: [12, 14]
    env:
      SKIP_PREFLIGHT_CHECK: true
    steps:
    - uses: actions/checkout@v2
    - uses: actions/setup-node@v1
      with: 
        node-version: ${{ matrix.node }}
    - name: Install dependencies
      uses: bahmutov/npm-install@v1
      with: 
        install-command: yarn
    - name: Run tests
      run: yarn test:ci