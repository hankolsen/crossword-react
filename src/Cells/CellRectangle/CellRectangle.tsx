import React from 'react';

import { CELL_HEIGHT, CELL_WIDTH } from 'config';

type Props = {
  column: number;
  row: number;
  highlighted?: boolean;
  selected?: boolean;
};

const CellRectangle = ({ column, row, highlighted = false, selected = false }: Props) => (
  <rect
    x={1 + (column * (CELL_WIDTH + 1))}
    y={1 + (row * (CELL_HEIGHT + 1))}
    width={CELL_WIDTH}
    height={CELL_HEIGHT}
    className={`crossword__cell ${highlighted ? 'crossword__cell--highlighted' : ''} ${selected ? 'crossword__cell--selected' : ''}`} // eslint-disable-line max-len
  />
);

export default CellRectangle;
