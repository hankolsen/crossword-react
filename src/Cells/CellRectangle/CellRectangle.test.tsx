import React from 'react';
import '@testing-library/jest-dom';
import { render } from 'test/test-utils';
import CellRectangle from 'Cells/CellRectangle/CellRectangle';

describe('CellLabel test', () => {
  it('should empty rectangle', async () => {
    const { container } = render(
      <svg>
        <CellRectangle row={1} column={1} />
      </svg>,
    );
    expect(container.getElementsByTagName('rect')[0]).toHaveClass(
      'crossword__cell',
    );
    expect(container.getElementsByTagName('rect')[0]).not.toHaveClass(
      'crossword__cell--highlighted',
    );
    expect(container.getElementsByTagName('rect')[0]).not.toHaveClass(
      'crossword__cell--selected',
    );
    expect(container).toMatchSnapshot();
  });

  it('should render selected rectangle', async () => {
    const { container } = render(
      <svg>
        <CellRectangle row={1} column={1} selected />
      </svg>,
    );
    expect(container.getElementsByTagName('rect')[0]).toHaveClass(
      'crossword__cell',
    );
    expect(container.getElementsByTagName('rect')[0]).not.toHaveClass(
      'crossword__cell--highlighted',
    );
    expect(container.getElementsByTagName('rect')[0]).toHaveClass(
      'crossword__cell--selected',
    );
    expect(container).toMatchSnapshot();
  });

  it('should render highlighted rectangle', async () => {
    const { container } = render(
      <svg>
        <CellRectangle row={1} column={1} highlighted />
      </svg>,
    );
    expect(container.getElementsByTagName('rect')[0]).toHaveClass(
      'crossword__cell',
    );
    expect(container.getElementsByTagName('rect')[0]).toHaveClass(
      'crossword__cell--highlighted',
    );
    expect(container.getElementsByTagName('rect')[0]).not.toHaveClass(
      'crossword__cell--selected',
    );
    expect(container).toMatchSnapshot();
  });

  it('should render highlighted and selected rectangle', async () => {
    const { container } = render(
      <svg>
        <CellRectangle row={1} column={1} highlighted selected />
      </svg>,
    );
    expect(container.getElementsByTagName('rect')[0]).toHaveClass(
      'crossword__cell',
    );
    expect(container.getElementsByTagName('rect')[0]).toHaveClass(
      'crossword__cell--highlighted',
    );
    expect(container.getElementsByTagName('rect')[0]).toHaveClass(
      'crossword__cell--selected',
    );
    expect(container).toMatchSnapshot();
  });
});
