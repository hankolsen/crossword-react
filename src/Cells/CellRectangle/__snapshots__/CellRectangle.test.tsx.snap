// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellLabel test should empty rectangle 1`] = `
<div>
  <div>
    <svg>
      <rect
        class="crossword__cell  "
        height="31"
        width="31"
        x="33"
        y="33"
      />
    </svg>
  </div>
</div>
`;

exports[`CellLabel test should render highlighted and selected rectangle 1`] = `
<div>
  <div>
    <svg>
      <rect
        class="crossword__cell crossword__cell--highlighted crossword__cell--selected"
        height="31"
        width="31"
        x="33"
        y="33"
      />
    </svg>
  </div>
</div>
`;

exports[`CellLabel test should render highlighted rectangle 1`] = `
<div>
  <div>
    <svg>
      <rect
        class="crossword__cell crossword__cell--highlighted "
        height="31"
        width="31"
        x="33"
        y="33"
      />
    </svg>
  </div>
</div>
`;

exports[`CellLabel test should render selected rectangle 1`] = `
<div>
  <div>
    <svg>
      <rect
        class="crossword__cell  crossword__cell--selected"
        height="31"
        width="31"
        x="33"
        y="33"
      />
    </svg>
  </div>
</div>
`;
